src/components/list/ItemsList.tsx

```diff
@@ -69,22 +84,29 @@ export default class ItemsList<T extends ListItem> extends React.Component<Props
     );
   }
 
-  private onSelectItem(item: T) {
-    const { selectedIds } = this.state;
+  private onSelectItem(item: T): void {
+    const { selectedItems } = this.state;
+    const { onSelect, select } = this.props;
+    const callback = onSelect ? () => onSelect([...this.state.selectedItems.values()]) : null;
```

Comment:
No need to send the values here, the keys should be sufficient.
Then you need a Set and not a  Map

Reply:
I need the objects here because I can't retrieve them from the Ids in the parent (Users) for the multi select (if I do a research for example, the state in Users will update and won't hold all the selected users) , and the objects are needed in ConnectorDetails (to display the users names)

Reply:
An alternative is to do an http request with the ids